tools/ci-operator-prowgen/main.go

```diff
@@ -0,0 +1,236 @@
+		},
+	}
+
+	postsubmit.Spec.Containers[0].Args = append(
+		postsubmit.Spec.Containers[0].Args,
+		fmt.Sprintf("--target=%s", test.Target),
+	)
+	postsubmit.Spec.Containers[0].Args = append(postsubmit.Spec.Containers[0].Args, "--promote")
+
+	return &postsubmit
+}
+
+func generateJobs(
+	configSpec *cioperatorapi.ReleaseBuildConfiguration,
+	org, repo, branch string,
+) (*map[string][]prowconfig.Presubmit, *map[string][]prowconfig.Postsubmit) {
+
+	orgrepo := fmt.Sprintf("%s/%s", org, repo)
+	presubmits := map[string][]prowconfig.Presubmit{}
+	postsubmits := map[string][]prowconfig.Postsubmit{}
+
+	imagesTest := false
+
+	for _, element := range configSpec.Tests {
+		// Check if config file has "images" test defined to avoid name clash
+		if element.As == "images" {
+			imagesTest = true
+		}
+		test := testDescription{Name: element.As, Target: element.As}
+		presubmits[orgrepo] = append(presubmits[orgrepo], *generatePresubmitForTest(test, org, repo, branch))
+		postsubmits[orgrepo] = append(postsubmits[orgrepo], *generatePostsubmitForTest(test, org, repo, branch))
+	}
+
+	if len(configSpec.Images) > 0 && !imagesTest {
+		// TODO: somehow handle the images case better than just not creating this job when there is name conflict
+		test := testDescription{Name: "images", Target: "[images]"}
+		presubmits[orgrepo] = append(presubmits[orgrepo], *generatePresubmitForTest(test, org, repo, branch))
+		postsubmits[orgrepo] = append(postsubmits[orgrepo], *generatePostsubmitForTest(test, org, repo, branch))
+	}
+
+	return &presubmits, &postsubmits
+}
+
+// these are unnecessary, and make the config larger so we strip them out
+func yamlBytesStripNulls(yamlBytes []byte) []byte {
+	nullRE := regexp.MustCompile("(?m)[\n]+^[^\n]+: null$")
+	return nullRE.ReplaceAll(yamlBytes, []byte{})
+}
+
+func generateProwJobsFromConfigFile(configFilePath string) []byte {
+	data, err := ioutil.ReadFile(configFilePath)
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "failed to read ci-operator config (%v)\n", err)
+		os.Exit(1)
+	}
+
+	var configSpec *cioperatorapi.ReleaseBuildConfiguration
+	if err := json.Unmarshal(data, &configSpec); err != nil {
+		fmt.Printf("failed to load ci-operator config (%v)\n", err)
+	}
+
+	configSpecDir := filepath.Dir(configFilePath)
+	repo := filepath.Base(configSpecDir)
+	org := filepath.Base(filepath.Dir(configSpecDir))
+	branch := strings.TrimSuffix(filepath.Base(configFilePath), filepath.Ext(configFilePath))
+
+	presubmits, postsubmits := generateJobs(configSpec, org, repo, branch)
+
+	jobConfig := prowconfig.JobConfig{
+		Presubmits:  *presubmits,
+		Postsubmits: *postsubmits,
+	}
+
+	jobConfigAsYaml, err := yaml.Marshal(jobConfig)
+	if err != nil {
+		fmt.Printf("failed to marshal the job config (%v)", err)
+		os.Exit(1)
+	}
+
+	jobConfigAsYaml = yamlBytesStripNulls(jobConfigAsYaml)
+
+	return jobConfigAsYaml
+}
+
+func generateAllProwJobs() {
+	repoRootRaw, err := exec.Command("git", "rev-parse", "--show-toplevel").Output()
```

Comment:
I was thinking that this is really a convenience tool and using it should be as simple as possible. The generator will almost always be called inside this repo and because we have a standard directory structure, everyone will always pass the same directories there. We can compute these defaults easily, why not do so?

I can of course add options to pass these manually, but I think we should keep the "easy" way, too.